geo-traits/src/rect.rs

```diff
@@ -0,0 +1,57 @@
+use geo_types::{Coord, CoordNum, Rect};
+
+use crate::{Dimension, PointTrait};
+
+/// A trait for accessing data from a generic Rect.
+pub trait RectTrait {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying coordinate, which implements [PointTrait]
+    type ItemType<'a>: 'a + PointTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimension;
+
+    /// The lower coordinate of this Rect
```

Comment:
"min"/"max" seems a little less ambiguous than "lower"/"upper"

e.g. a lot of graphics programming has (0, 0) as top left.